src/sentry/notifications/helpers.py

```diff
@@ -610,3 +641,149 @@ def get_providers_for_recipient(
+        return True
+    return isinstance(recipient, (RpcUser, User))
+
+
+def recipient_is_team(recipient: RpcActor | Team | User) -> bool:
+    from sentry.models.team import Team
+
+    if isinstance(recipient, RpcActor) and recipient.actor_type == ActorType.TEAM:
+        return True
+    return isinstance(recipient, Team)
+
+
+def get_all_setting_providers(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> Iterable[NotificationSettingProvider]:
+    project_settings = (
+        Q(scope_type=NotificationScopeType.PROJECT.value, scope_identifier=project.id)
+        if project
+        else Q()
+    )
+
+    org_settings = (
+        Q(
+            scope_type=NotificationScopeType.ORGANIZATION.value,
+            scope_identifier=project.organization.id,
+        )
+        if organization
+        else Q()
+    )
+
+    team_or_user_settings = Q()
+    if recipient_is_user(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.USER.value, scope_identifier=recipient.id
+        )
+    elif recipient_is_team(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.TEAM.value, scope_identifier=recipient.id
+        )
+
+    return NotificationSettingProvider.objects.filter(
+        project_settings | org_settings | team_or_user_settings
+    )
+
+
+def get_all_setting_options(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+):
+    project_settings = (
+        Q(scope_type=NotificationScopeType.PROJECT.value, scope_identifier=project.id)
+        if project
+        else Q()
+    )
+
+    org_settings = (
+        Q(
+            scope_type=NotificationScopeType.ORGANIZATION.value,
+            scope_identifier=project.organization.id,
+        )
+        if organization
+        else Q()
+    )
+
+    team_or_user_settings = Q()
+    if recipient_is_user(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.USER.value, scope_identifier=recipient.id
+        )
+    elif recipient_is_team(recipient):
+        team_or_user_settings = Q(
+            scope_type=NotificationScopeType.TEAM.value, scope_identifier=recipient.id
+        )
+
+    NotificationSettingOption.objects.filter(
+        project_settings | org_settings | team_or_user_settings
+    )
+
+
+def get_settings_for_recipient(
+    recipient: RpcActor | Team | User,
+    project: Project | None = None,
+    organization: Organization | None = None,
+) -> MutableMapping[NotificationScopeType, NotificationSettingOptionValues]:
+    all_settings = get_all_setting_options(recipient, project)
+
+    notification_settings = {}
+    # Project settings take precedence over all other notification settings
+    for setting in all_settings:
+        if setting.scope_type == NotificationScopeType.PROJECT.value:
```

Comment:
@snigdhas If a setting has the `scope_type` of project, we probably should check the org settings as well. In practice, we only set one or the other but if we want to enable layered settings down the line then I think we should check the project and org.

Reply:
The intention behind this code block is to add the layered settings - We start by checking project settings, then add on the org scope IF there isn't already a setting for the project and then do the same for team/user. I'll add some tests to confirm.